common/src/WooCommerce.php

```diff
@@ -12,17 +12,23 @@ class WooCommerce {
 
     public function __construct(BasePlugin $plugin) {
         $this->plugin = $plugin;
-        add_action('woocommerce_order_status_completed', [$this, 'order_completed'], 10, 1);
+        add_action('woocommerce_order_status_changed', [$this, 'orderStatusChanged'], 10, 3);
         add_action('woocommerce_checkout_update_order_meta', [$this, 'set_order_language']);
     }
 
+    public function orderStatusChanged(int $order_id, string $old_status, string $new_status): void {
+        if ('wc-' . $new_status == (string) get_option($this->plugin->getOptionName('order_status'))) {
```

Comment:
It seems like it is possible to create custom statuses without the `wc-` prefix, using [this plugin](https://woocommerce.com/products/woocommerce-order-status-manager/), for example.  WooCommerce removes the `wc-` prefix if it is there, but it's not guaranteed to be there.

So perhaps this would be better written as:

~~~
in_array(get_option($this->plugin->getOptionName('order_status'), [$new_status, 'wc-' . $new_status])
~~~

or

~~~
$new_status == preg_replace('/^wc-/', '', get_option($this->plugin->getOptionName('order_status')))
~~~

**Secondly,** you need to make sure that when the plugin is updated it doesn't suddenly break the invites. In this case it will, because the `order_status` option is not defined. So better make sure that if the option doesn't exist/is empty, you use the default value.

**Lastly,** but this is up to you, the OpenCart and PrestaShop plugins allow selecting multiple statuses, so that if not all orders are handled in the same way, you can still send invites for both statuses. That might be worth considering.

PS: What's the reason for using `(string)` here?

Reply:
👍 I went for your suggestion for having multiple statuses, similar to opencart3.